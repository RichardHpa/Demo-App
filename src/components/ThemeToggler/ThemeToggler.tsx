import { IconButton } from 'components/Button'

import { useTheme, Theme } from 'providers/ThemeProvider'

import { SunIcon, MoonIcon } from '@heroicons/react/24/solid'

export const ThemeToggler = () => {
  const [theme, setTheme] = useTheme()

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme === Theme.LIGHT ? Theme.DARK : Theme.LIGHT))
  }
  return (
    <IconButton
      variant='ghost'
      color='inherit'
      icon={theme === Theme.LIGHT ? <MoonIcon /> : <SunIcon />}
      onClick={toggleTheme}
      aria-label={theme === Theme.LIGHT ? 'Switch to dark mode' : 'Switch to light mode'}
    />
  )
}
