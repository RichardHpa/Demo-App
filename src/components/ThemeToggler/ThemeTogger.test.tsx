import { setMatchMediaMock } from 'tests/matchMediaMock'
import { render, screen } from 'tests/renderHelpers'
import userEvent from '@testing-library/user-event'

import { ThemeToggler } from './ThemeToggler'

describe('ThemeToggler', () => {
  beforeEach(() => {
    setMatchMediaMock()
  })

  test('renders ThemeToggler with default dark mode', () => {
    render(<ThemeToggler />)
    const button = screen.getByRole('button', { name: /switch to dark mode/i })

    expect(button).toBeInTheDocument()
  })

  test('clicks on the button changes to theme', () => {
    render(<ThemeToggler />)
    const button = screen.getByRole('button', { name: /switch to dark mode/i })
    expect(button).toBeInTheDocument()
    expect(screen.queryByRole('button', { name: /switch to light mode/i })).not.toBeInTheDocument()

    userEvent.click(button)

    const newButton = screen.getByRole('button', { name: /switch to light mode/i })
    expect(screen.queryByRole('button', { name: /switch to dark mode/i })).not.toBeInTheDocument()
    expect(newButton).toBeInTheDocument()
  })

  test('renders ThemeToggler snapshot', () => {
    const { container } = render(<ThemeToggler />)
    expect(container).toMatchSnapshot()
  })
})
