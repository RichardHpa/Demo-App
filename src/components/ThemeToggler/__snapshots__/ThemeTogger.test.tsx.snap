// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggler renders ThemeToggler with default dark mode 1`] = `
<button
  aria-label="Switch to dark mode"
  class="inline-flex align-middle justify-center rounded-full p-2 text-blue-500 hover:bg-blue-500 hover:text-white border-transparent hover:bg-opacity-25"
>
  <svg
    aria-hidden="true"
    class="h-6 w-6 inline-block"
    fill="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.528 1.718a.75.75 0 01.162.819A8.97 8.97 0 009 6a9 9 0 009 9 8.97 8.97 0 003.463-.69.75.75 0 01.981.98 10.503 10.503 0 01-9.694 6.46c-5.799 0-10.5-4.701-10.5-10.5 0-4.368 2.667-8.112 6.46-9.694a.75.75 0 01.818.162z"
      fill-rule="evenodd"
    />
  </svg>
</button>
`;
